src/index.ts

```diff
@@ -18,6 +24,27 @@ export interface JsonFetchOptions extends Omit<RequestInit, 'body'> {
   retry?: Parameters<typeof promiseRetry>[0];
   timeout?: number;
   expectedStatuses?: Array<number>;
+  onRequestStart?: ({
+    agent,
+    body,
+    shouldRetry,
```

Comment:
I don't think this is useful here? It's a function and it doesn't really make sense before the request anyway.